server/routes/auth.ts

```diff
@@ -40,121 +41,63 @@ app.use(session({
-	router.post("/confirm", validateAndCacheHostName, postParser, async (request, response) => {
-		let user = request.user as IUserMongoose;
-		let name = request.body.name as string;
-		if (!name || !name.trim()) {
-			request.flash("error", "Invalid name");
-			response.redirect("/login/confirm");
-			return;
-		}
-		if (!request.isAuthenticated() || !user) {
-			request.flash("error", "Must be logged in");
-			response.redirect("/login");
-			return;
-		}
-		user.name = name.trim();
-
-		let email = request.body.email as string | undefined;
-		if (email && email !== user.email) {
-			if (!email.trim()) {
-				request.flash("error", "Invalid email");
-				response.redirect("/login/confirm");
-				return;
-			}
-			if (await User.count({ email }) > 0) {
-				request.flash("error", "That email address is already in use. You may already have an account from another login service.");
-				response.redirect("/login/confirm");
-				return;
-			}
-			user.admin = false;
-			user.verifiedEmail = false;
-			user.email = email;
-			if (config.admins.includes(email)) {
-				user.admin = true;
-				console.info(`Adding new admin: ${email}`);
-			}
-		}
-		user.accountConfirmed = true;
+authRoutes.get("/validatehost/:nonce", (request, response) => {
+	let nonce: string = request.params.nonce || "";
+	response.send(crypto.createHmac("sha256", config.secrets.session).update(nonce).digest().toString("hex"));
+});
 
-		try {
-			await user.save();
-			if (!user.verifiedEmail && (!user.local || !user.local.verificationCode)) {
-				await sendVerificationEmail(request, user);
-			}
-			if (!user.verifiedEmail) {
-				request.logout();
-				request.flash("success", "Account created successfully. Please verify your email before logging in.");
-				response.redirect("/login");
-				return;
+authRoutes.all("/logout", (request, response) => {
+	let user = request.user as IUser | undefined;
+	if (user) {
+		let groundTruthURL = new URL(config.secrets.groundTruth.url);
+		let requester = groundTruthURL.protocol === "http:" ? http : https;
+		requester.request(new URL("/api/user/logout", config.secrets.groundTruth.url), {
+			method: "POST",
+			headers: {
+				"Authorization": `Bearer ${user.token}`
 			}
-			response.redirect("/");
-		}
-		catch (err) {
-			console.error(err);
-			request.flash("error", "An error occurred while creating your account");
-			response.redirect("/login/confirm");
-		}
-	});
+		}).end();
 
-	router.get("/validatehost/:nonce", (request, response) => {
-		let nonce: string = request.params.nonce || "";
-		response.send(crypto.createHmac("sha256", config.secrets.session).update(nonce).digest().toString("hex"));
-	});
-
-	router.all("/logout", (request, response) => {
 		request.logout();
-		response.redirect("/login");
-	});
-}
-reloadAuthentication().catch(err => {
-	throw err;
+	}
+	if (request.session) {
+		request.session.loginAction = "render";
+	}
+	response.redirect("/login");
```

Comment:
Should we redirect user to registration login or ground truth login

Reply:
Would GT know to redirect back?

Reply:
It redirects to `/login` so that we can tell the user that they've successfully logged out. If we redirect to Ground Truth, the user will be told to login again even if that isn't what they want to do